### ISSUE TITLE
get TypeError: 'list' object is not callable when running up with ipdb

### ISSUE THREAD

**Issue by nathanielatom** [gotcha/ipdb#210]
Hi, 

I get the following error when trying the `up` command with ipdb:

```
In [1]: def foo(): 
   ...:     return bar() 
   ...:                                                                                                                                        

In [2]: def bar(): 
   ...:     raise TypeError 
   ...:                                                                                                                                        

In [3]: foo()                                                                                                                                  
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-c19b6d9633cf> in <module>
----> 1 foo()

<ipython-input-1-c5d436c2278c> in foo()
      1 def foo():
----> 2     return bar()
      3 

<ipython-input-2-e5e1cb352776> in bar()
      1 def bar():
----> 2     raise TypeError
      3 

TypeError: 

In [4]: %debug                                                                                                                                 
> <ipython-input-2-e5e1cb352776>(2)bar()
      1 def bar():
----> 2     raise TypeError
      3 

ipdb> up                                                                                                                                       
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-bc99e4ec804d> in <module>
----> 1 get_ipython().run_line_magic('debug', '')

~/.miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py in run_line_magic(self, magic_name, line, _stack_depth)
   2325                 kwargs['local_ns'] = self.get_local_scope(stack_depth)
   2326             with self.builtin_trap:
-> 2327                 result = fn(*args, **kwargs)
   2328             return result
   2329 

<decorator-gen-57> in debug(self, line, cell)

~/.miniconda/lib/python3.7/site-packages/IPython/core/magic.py in <lambda>(f, *a, **k)
    185     # but it's overkill for just that one bit of state.
    186     def magic_deco(arg):
--> 187         call = lambda f, *a, **k: f(*a, **k)
    188 
    189         if callable(arg):

~/.miniconda/lib/python3.7/site-packages/IPython/core/magics/execution.py in debug(self, line, cell)
    467 
    468         if not (args.breakpoint or args.statement or cell):
--> 469             self._debug_post_mortem()
    470         elif not (args.breakpoint or cell):
    471             # If there is no breakpoints, the line is just code to execute

~/.miniconda/lib/python3.7/site-packages/IPython/core/magics/execution.py in _debug_post_mortem(self)
    481 
    482     def _debug_post_mortem(self):
--> 483         self.shell.debugger(force=True)
    484 
    485     def _debug_exec(self, code, breakpoint):

~/.miniconda/lib/python3.7/site-packages/IPython/core/interactiveshell.py in debugger(self, force)
   1190             return
   1191 
-> 1192         self.InteractiveTB.debugger(force=True)
   1193 
   1194     #-------------------------------------------------------------------------

~/.miniconda/lib/python3.7/site-packages/IPython/core/ultratb.py in debugger(self, force)
   1265                     etb = etb.tb_next
   1266                 self.pdb.botframe = etb.tb_frame
-> 1267                 self.pdb.interaction(None, etb)
   1268 
   1269         if hasattr(self, 'tb'):

~/.miniconda/lib/python3.7/site-packages/IPython/core/debugger.py in interaction(self, frame, traceback)
    317     def interaction(self, frame, traceback):
    318         try:
--> 319             OldPdb.interaction(self, frame, traceback)
    320         except KeyboardInterrupt:
    321             self.stdout.write("\n" + self.shell.get_exception_only())

~/.miniconda/lib/python3.7/site-packages/_pdbpp_path_hack/pdb.py in interaction(self, frame, traceback)
    229         # Use _cmdloop on py3 which catches KeyboardInterrupt.
    230         if hasattr(self, '_cmdloop'):
--> 231             self._cmdloop()
    232         else:
    233             self.cmdloop()

~/.miniconda/lib/python3.7/pdb.py in _cmdloop(self)
    317                 # the current command, so allow them during interactive input
    318                 self.allow_kbdint = True
--> 319                 self.cmdloop()
    320                 self.allow_kbdint = False
    321                 break

~/.miniconda/lib/python3.7/site-packages/IPython/terminal/debugger.py in cmdloop(self, intro)
    134 
    135                 line = self.precmd(line)
--> 136                 stop = self.onecmd(line)
    137                 stop = self.postcmd(stop, line)
    138             self.postloop()

~/.miniconda/lib/python3.7/pdb.py in onecmd(self, line)
    414         """
    415         if not self.commands_defining:
--> 416             return cmd.Cmd.onecmd(self, line)
    417         else:
    418             return self.handle_command_def(line)

~/.miniconda/lib/python3.7/cmd.py in onecmd(self, line)
    215             except AttributeError:
    216                 return self.default(line)
--> 217             return func(arg)
    218 
    219     def emptyline(self):

~/.miniconda/lib/python3.7/site-packages/IPython/core/debugger.py in do_up(self, arg)
    727             _newindex = self.curindex
    728             counter = 0
--> 729             hidden_frames = self.hidden_frames(self.stack)
    730             for i in range(self.curindex - 1, -1, -1):
    731                 frame = self.stack[i][0]

TypeError: 'list' object is not callable

In [5]:  
```

Sorry, I'm not totally clear whether this should be tracked here or in ipython itself.

ipdb v0.13.4
ipython v7.19.0
python v3.7.6

Thanks!

**Comment by patricknraanes** [comment]
No idea why, but this was fixed for me by not installing pdbpp.

Btw, I'm on Mac.

**Comment by DerWeh** [comment]
I can confirm, that uninstalling `pdbpp` fixes this issue.

**Comment by nathanielatom** [comment]
Yep, I confirmed uninstalling `pdbpp` fixes this issue on both mac and linux. I have `pdbpp v0.10.2`.

**Comment by nathanielatom** [comment]
Issue is resolved in master branch of pdbpp: pdbpp/pdbpp#408

**Comment by haoyangrice** [comment]
Very helpful! I can confirm that uninstalling `pdbpp` fixed the issue on Windows 10 as well.
